core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,525 @@
+    await this.clickOn(deleteConfirmButton);
+  }
+
+  /**
+   * This function helps in verifying , if exploration is
+   * deleted successfully?
+   */
+  async expectExplorationToBeDeletedSuccessfullyFromCreatorDashboard(
+  ): Promise<void> {
+    await this.page.waitForTimeout(500);
+    try {
+      await this.page.goto(explorationUrlAfterPublished);
+      throw new Error('Exploration is not deleted successfully.');
+    } catch (error) {
+      showMessage('Exploration is Successfully deleted.');
+    }
+  }
+
+  /**
+   * This function helps in drafting the exploration.
+   */
+  async saveDraftExploration(): Promise<void> {
+    await this.clickOn(saveDraftButton);
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn('.e2e-test-save-draft-button');
+  }
+
+  /**
+   * This function checks whether changes has been drafted or not.
+   */
+  async expectExplorationToBeDraftedSuccessfully(): Promise<void> {
+    const button = await this.page.$('#tutorialSaveButton');
+
+    if (!button) {
+      throw new Error('Save button not found.');
+    }
+
+    const isDraftButtonDisabled = await button.evaluate(
+      (button) => (button as HTMLButtonElement).disabled);
+
+    if (isDraftButtonDisabled) {
+      showMessage('Changes have been successfully drafted.');
+    } else {
+      throw new Error('Failed to draft changes.');
+    }
+  }
+
+  /**
+   * This function helps in publishing the exploration
+   */
+  async publishExploration(): Promise<void> {
+    await this.saveDraftExploration();
+    await this.page.waitForTimeout(500);
+    await this.clickOn(publishButton);
+    await this.clickOn(publishConfirmButton);
+    await this.clickOn('.e2e-test-confirm-publish');
+    await this.clickOn(closePublishedPopUp);
+
+    await this.page.waitForTimeout(500);
+    explorationUrlAfterPublished = await this.page.url();
+  }
+
+  /**
+  *This function checks whether the exploration
+  *is published successfully or not.
+  */
+  async expectInteractionOnCreatorDashboard(): Promise<void> {
+    try {
+      await this.page.goto(explorationUrlAfterPublished);
+      showMessage('Exploration is available on creator dashboard.');
+    } catch (error) {
+      throw new Error('Failed to navigate to the exploration URL.');
+    }
+  }
+
+  /**
+   * This function is discarding the current changes.
+   */
+  async discardCurrentChanges(): Promise<void> {
+    await this.clickOn('.e2e-test-settings-container');
+    await this.clickOn('button.e2e-test-save-discard-toggle');
+    await this.page.waitForTimeout(500);
+    await this.clickOn(discardDraftButton);
+    await this.clickOn(discardConfirmButton);
+
+    await this.page.waitForNavigation({ waitUntil: 'networkidle2' });
+  }
+
+  /**
+  *This function checks whether changes has discarded successfully or not.
+  */
+  async expectTitleToBe(titleBeforeChanges: string): Promise<void> {
+    await this.page.waitForTimeout(400);
```

Comment:
The presence of a `waitForTimeout` statement usually signifies a lack of clarity of what state the UI should be in before interacting with it. If removing it causes the test to timeout, then it is worth investigating why the test is timing out.